nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/cloudwatch/PutCloudWatchMetric.java

```diff
@@ -70,6 +70,33 @@
     public static final Set<Relationship> relationships = Collections.unmodifiableSet(
             new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE)));
 
+    private static final Set<String> units = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    "Seconds", "Microseconds", "Milliseconds", "Bytes",
+                    "Kilobytes", "Megabytes", "Gigabytes", "Terabytes",
+                    "Bits", "Kilobits", "Megabits", "Gigabits", "Terabits",
+                    "Percent", "Count", "Bytes/Second", "Kilobytes/Second",
+                    "Megabytes/Second", "Gigabytes/Second", "Terabytes/Second",
+                    "Bits/Second", "Kilobits/Second", "Megabits/Second",
+                    "Gigabits/Second", "Terabits/Second", "Count/Second",
+                    "None", "")));
```

Comment:
The Unit field is not a hard requirement for the API request, so I included an empty string as a check so the validator wouldn't complain if the user didnt provide one.

Reply:
The API docs outline the appropriate values: https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html

Reply:
I've seen this documentation, that's why I'm asking it. I think if you don't need Unit for the request you need to set that property to null, not empty string. Since the required property for the PropertyDescriptor is false, the validator will only run when a value is set for that field. Though I just realized that maybe for compatibility reasons we need to allow empty string, because previously everything was allowed.

Reply:
How would you guys like to proceed here? I think null will take some tweaking to prevent issues with the property value. @nandorsoma @exceptionfactory

Reply:
Following up on this discussion, it seems better to check for an empty string separately from the list of allowed types. Taking another look at this, do the valid values exist anywhere in the AWS SDK? It seems like this list could become difficult to maintain.

Reply:
It's a java enum -> https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/cloudwatch/model/StandardUnit.html

we could potentially leverage the SDK and call the types from the software.amazon.awssdk.services.cloudwatch.model package to prevent breaking changes.

Reply:
Thanks for the reference @patalwell! Leveraging the enum values sounds like the optimal way forward.